src/buildblock/ProjDataInfoSubsetByView.cxx

```diff
@@ -0,0 +1,202 @@
+//
+//
+/*
+    Copyright (C) 2021, University College London
+    This file is part of STIR.
+
+    SPDX-License-Identifier: Apache-2.0
+    See STIR/LICENSE.txt for details
+*/
+/*!
+  \file
+  \ingroup projdata
+
+  \brief Implementation of non-inline functions of class stir::ProjDataInfoSubsetByView
+
+  \author Kris Thielemans
+
+*/
+#include "stir/ProjDataInfoSubsetByView.h"
+#include "stir/Bin.h"
+#include "stir/Array.h"
+
+START_NAMESPACE_STIR
+
+ProjDataInfoSubsetByView::ProjDataInfoSubsetByView(const shared_ptr<const ProjDataInfo> full_proj_data_info_sptr,
+                                                   const std::vector<int>& views)
+  :
+  ProjDataInfo(full_proj_data_info_sptr->get_scanner_sptr(),
+               VectorWithOffset<int>(full_proj_data_info_sptr->get_min_segment_num(),
+                                     full_proj_data_info_sptr->get_max_segment_num()), // filled in below
+               views.size(),
+               full_proj_data_info_sptr->get_num_tangential_poss()),
+  org_proj_data_info_sptr(full_proj_data_info_sptr->clone()),
```

Comment:
Is it possible this is the cause of our segfault? ProjDataInfoCylindricalNoArcCorr doesn't have a copy constructor (called by clone) implemented, nor does ProjDataInfo. But ProjDataInfo has shared_ptr members. Are these being shallow copied?

Reply:
In which case we've uncovered a bug with `ProjDataInfo`? But also is it surprising to the user that if you creat a `ProjDataInfoSubsetByView` and then modify it, the `org_proj_data_info_sptr` changes, but the original `proj_data_info_sptr` doesn't.

Reply:
@rijobro has introduced a lot of `shared_ptr<const A>` in arguments to prevent this trouble with these things. That means introducing more `clone()`, which is less efficient, but does prevent surprising things like modifications of one object affecting another one. I'm not 100% sure of course if we've caught all of these.

In any case, I didn't think that in this test code we are modifying such settings. And even if we did, that wouldn't cause a segfault as far as I can see, just surprising behaviour.